api-test/internal/wsotp/ws.go

```
@@ -0,0 +1,406 @@
+	// Wait for the frame to start. Nothing has been consumed yet, so a timeout
+	// here leaves the stream exactly where it was.
+	if resumable {
+		c.setReadDeadline(c.idleTimeout)
+	} else {
+		c.setReadDeadline(c.frameTimeout)
+	}
+	if _, err = c.br.Peek(1); err != nil {
+		if resumable {
+			return false, 0, nil, err
+		}
+		return false, 0, nil, fmt.Errorf("%w: %v", errStreamDesync, err)
```

golangci-lint (2.12.2)

[error] non-wrapping format verb for fmt.Errorf. Use `%w` to format errors

(errorlint)